pkg/cli/utils/apimodel.go

```diff
@@ -40,33 +39,27 @@ import (
 	"github.com/openclarity/vmclarity/pkg/shared/utils"
 )
 
-func ConvertSBOMResultToAPIModel(sbomResults *sbom.Results) *models.SbomScan {
+func ConvertSBOMResultToPackages(sbomResults *sbom.Results) *[]models.Package {
 	packages := []models.Package{}
 
 	if sbomResults.SBOM.Components != nil {
 		for _, component := range *sbomResults.SBOM.Components {
-			packages = append(packages, *ConvertPackageInfoToAPIModel(component))
+			packages = append(packages, models.Package{
+				Cpes:     utils.PointerTo([]string{component.CPE}),
+				Language: utils.PointerTo(cyclonedx_helper.GetComponentLanguage(component)),
+				Licenses: utils.PointerTo(cyclonedx_helper.GetComponentLicenses(component)),
+				Name:     utils.PointerTo(component.Name),
+				Purl:     utils.PointerTo(component.PackageURL),
+				Type:     utils.PointerTo(string(component.Type)),
+				Version:  utils.PointerTo(component.Version),
+			})
 		}
 	}
 
-	return &models.SbomScan{
-		Packages: &packages,
-	}
+	return &packages
 }
 
-func ConvertPackageInfoToAPIModel(component cdx.Component) *models.Package {
-	return &models.Package{
-		Cpes:     utils.PointerTo([]string{component.CPE}),
-		Language: utils.PointerTo(cyclonedx_helper.GetComponentLanguage(component)),
-		Licenses: utils.PointerTo(cyclonedx_helper.GetComponentLicenses(component)),
-		Name:     utils.PointerTo(component.Name),
-		Purl:     utils.PointerTo(component.PackageURL),
-		Type:     utils.PointerTo(string(component.Type)),
-		Version:  utils.PointerTo(component.Version),
-	}
-}
-
-func ConvertVulnResultToAPIModel(vulnerabilitiesResults *vulnerabilities.Results) *models.VulnerabilityScan {
+func ConvertVulnResultToVulnerabilities(vulnerabilitiesResults *vulnerabilities.Results) *[]models.Vulnerability {
 	// nolint:prealloc
 	var vuls []models.Vulnerability
 	for _, vulCandidates := range vulnerabilitiesResults.MergedResults.MergedVulnerabilitiesByKey {
```

Comment:
I know that this is not in the scope of this PR, but don't we need to check for `nil` before looping here? Same for malware and rootkits below? It seems like some methods check for nil and others do not

Reply:
I added `nil` checking for every method here.